test/rules/promise-async/test.ts.lint

```diff
@@ -0,0 +1,38 @@
+const nonAsyncPromiseFunction = function(p: Promise<void>) { return p; };
```

Comment:
good question... I don't think this has come up before. perhaps we can add a `"compilerOptions"` block to our test `tslint.json` files?

N.B. as you can see from this thread (https://github.com/palantir/tslint/issues/1445), it might be a good idea to put `"compilerOptions"` under a namespace in the test `tslint.json` so as to clearly indicate it is a private API for use in this repo only and not a public API for tslint users yet.